cmd/kube-proxy/app/server_test.go

```diff
@@ -272,23 +326,82 @@ func Test_detectNodeIPs(t *testing.T) {
 				netutils.ParseIPSloppy("90.90.90.90"),
 				netutils.ParseIPSloppy("2001:db8::2"),
 			},
-			bindAddress:    "::1",
+			nodeIPOverride: []string{"::1"},
 			expectedFamily: v1.IPv6Protocol,
+			ipFamilyPolicy: v1.IPFamilyPolicyPreferDualStack,
 			expectedIPv4:   "90.90.90.90",
 			expectedIPv6:   "::1",
 		},
+		{
+			name: "Dual stack, override with unspecified addresses, IPv6",
+			rawNodeIPs: []net.IP{
+				netutils.ParseIPSloppy("2001:db8::2"),
+				netutils.ParseIPSloppy("90.90.90.90"),
+			},
+			nodeIPOverride: []string{"0.0.0.0", "::"},
+			expectedFamily: v1.IPv6Protocol,
```

Comment:
Uh... this seems wrong. If the user says `nodeIPOverride: [ "0.0.0.0", "::" ]`, then they are saying they want IPv4-primary.

It seems this is consistent with the existing `--bind-address` behavior though. Ah, but that's probably because there's no such thing as `bindAddress: nil`; it defaults to `"0.0.0.0"`, so we can't treat `"0.0.0.0"` as being an actual opinion from the user.

Ugh... ok, I want to think about this again. I think we need to just have different, incompatible node-IP-detecting behavior for v1alpha1 and v1alpha2 configs.

Reply:
Sorry, I never got back to this. So, I think the fix for now is to just have both `bindAddress` and `nodeIPOverride` in the internal version of the config, where only `bindAddress` would get filled in for CLI args or v1alpha1 config, and only `nodeIPOverride` would get filled in for v1alpha2 config. Then the kube-proxy startup code can use the backward-compatible behavior for CLI/v1alpha1 configs, and the better, more-explicit behavior for v1alpha2 configs